src/prefect/futures.py

```diff
@@ -31,22 +35,39 @@
 class PrefectFuture(abc.ABC, Generic[R]):
     """
     Abstract base class for Prefect futures. A Prefect future is a handle to the
-    asynchronous execution of a task run. It provides methods to wait for the task
-    to complete and to retrieve the result of the task run.
+    asynchronous execution of a run. It provides methods to wait for the
+    to complete and to retrieve the result of the run.
     """
 
     def __init__(self, task_run_id: uuid.UUID):
+        warnings.warn(
+            "The __init__ method of PrefectFuture is deprecated and will be removed in a future release. "
+            "If you are subclassing PrefectFuture, please implement the __init__ method in your subclass.",
+            DeprecationWarning,
+        )
         self._task_run_id = task_run_id
-        self._final_state: Optional[State[R]] = None
+        self._final_state: State[R] | None = None
 
     @property
     def task_run_id(self) -> uuid.UUID:
         """The ID of the task run associated with this future"""
+        warnings.warn(
+            "The task_run_id property of PrefectFuture is deprecated and will be removed in a future release. "
+            "If you are subclassing PrefectFuture, please implement the task_run_id property in your subclass.",
+            DeprecationWarning,
+        )
+
         return self._task_run_id
 
     @property
     def state(self) -> State:
         """The current state of the task run associated with this future"""
+        warnings.warn(
+            "The state property of PrefectFuture is deprecated and will be removed in a future release. "
+            "If you are subclassing PrefectFuture, please implement the state property in your subclass.",
```

Comment:
is it worth pointing these folks to the new class?

Reply:
and is the plan for these to raise `NotImplemented` errors in the future?

Reply:
Yeah, I can point to the new class. The plan is to remove these methods from `PrefectFuture` and let subclasses manage their implementation.

Reply:
Updated in [68fcc8d](https://github.com/PrefectHQ/prefect/pull/17234/commits/68fcc8d7ca7412adfdcb3d79ec9000c4b65c5cff)